### PR TITLE
remove null values when we turn the group_by into an array

### PR DESCRIPTION
**Pull request by josephwegner**
## What does this PR do? How does it affect users?
Saved queries will represent `group_by` as a `null` for queries without a group_by value. Explorer tries to be smart by turning non-array `group_by` values into single-element arrays, but makes the poor mistake of wrapping `null` `group_by`s into an array. That messes things up down the line, specifically causing Explorer to think the query is categorial (has a group_by).

This change removes `null` values from the `group_by` array. Effectively, for users, that means they will get the correct chart type options for their `singular` saved queries.

## How should this be tested?

Pull it down, and create a regular ol' count query with a metric type. Save it. The chart type should still be metric. If it's not we have trouble.

I'd also suggest trying a few other query types, just to watch for unintended consequences.

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:

- [ ] Does the code compile without warnings (check shell, console)?
- [ ] Do all tests pass?
- [ ] Does the UI, pixel by pixel, look exactly as expected (check various screen sizes, including mobile)?
- [ ] If the feature makes requests from the browser, inspect them in the Web Inspector. Do they look as expected (parameters, headers, etc)?
- [ ] If the feature sends data to Keen, is the data visible in the project if you run an extraction (include link to collection/query)?
- [ ] If the feature saves data to a database, can you confirm the data is indeed created in the database?

## Related tickets?
#192 (JIRA #520)

### CODE REVIEW

**Comment by josephwegner** [comment]
Oh, I forgot to mention, I pinned our `keen-analysis` to 1.1.0... Dustin has a bug that he's working on in the recently released 1.2.0 that was causing test failures.

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "classnames": "2.1.2",
     "flux": "^2.0.3",
     "json-stable-stringify": "1.0.0",
-    "keen-analysis": "^1.1.0",
+    "keen-analysis": "1.1.0",
```
Comment (sfalkoff) [inline package.json]: Thanks for the heads up about the keen-analysis version. This just in: looks like it's being addressed in https://github.com/keen/explorer/pull/197.

Comment (josephwegner) [inline package.json]: @sfalkoff His PR should overwrite my pinning, so I'm gonna leave this in (that will avoid CI/build errors depending on which merges first)
